File: taller_hilos/ejercicio6/ejercicio6.py
import threading
import string

# Ejercicio 6: Hilo que imprime letras de la A a la Z
#
# string.ascii_uppercase es la cadena 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.
# Iterar sobre ella es más legible que usar códigos ASCII numéricos.

def imprimir_letras():
    # Cada iteración entrega una letra del alfabeto en mayúsculas.
    for letra in string.ascii_uppercase:
        print(f"Letra: {letra}")

hilo = threading.Thread(target=imprimir_letras)
hilo.start()
hilo.join()  # Esperamos a que el hilo termine de imprimir todas las letras.

File: taller_hilos/ejercicio6/Ejercicio6.java
// Ejercicio 6: Hilo que imprime letras de la A a la Z
//
// En Java los char son enteros, por lo que se puede iterar
// incrementando la variable como si fuera un número.
// El hilo secundario se encarga de recorrer todo el alfabeto.

class ImpresionLetras implements Runnable {
    public void run() {
        // 'A' vale 65 y 'Z' vale 90 en Unicode/ASCII.
        // El bucle avanza de carácter en carácter hasta Z.
        for (char c = 'A'; c <= 'Z'; c++) {
            System.out.println("Letra: " + c);
        }
    }
}

public class Ejercicio6 {
    public static void main(String[] args) {
        Thread hilo = new Thread(new ImpresionLetras());
        // start() delega la impresión de letras al hilo secundario.
        hilo.start();
    }
}
